### PR TITLE
e2e: add negative tests for duplicate primary UDN/CUDN in same namespace

### PR DESCRIPTION
**Pull request by jechen0648**
Add negative e2e coverage for the rule that only one primary network may exist per namespace. These new tests create a ready primary UDN or CUDN first, then verify a second primary network in the same namespace is rejected with the expected status condition.

## 📑 Description

- Add an e2e negative test that creates a ready primary `UserDefinedNetwork`, then verifies a second primary UDN in the same namespace reports `NetworkCreated=False` with `SyncError`.
- Add an e2e negative test that creates a ready primary `ClusterUserDefinedNetwork`, then verifies a second primary CUDN targeting the same namespace reports `NetworkCreated=False` with `NetworkAttachmentDefinitionSyncError`.
- Both tests also verify the first network remains ready and the second network does not create a `NetworkAttachmentDefinition`.

This closes a coverage gap: existing negative tests only checked the case where a standalone primary NAD already existed before creating a UDN/CUDN.

Test plan
 Tested locally on KIND with network segmentation enabled

Fixes #

## Release Notes

```release-note
NONE
```

## ✅ Checks

- [ ] My code requires changes to the documentation
- [ ] if so, I have updated the documentation as required
- [ ] My code does not require changes to the documentation
- [ ] My code requires tests
- [ ] if so, I have added and/or updated the tests as required
- [ ] My code does not require tests
- [ ] All the tests have passed in the CI

### CODE REVIEW

File: test/e2e/network_segmentation.go [hunk 1]
```diff
@@ -1704,6 +1704,46 @@ spec:
...
+			return normalizeConditions(actualConditions)
+		}, 5*time.Second, 1*time.Second).Should(ConsistOf(metav1.Condition{
+			Type:    "NetworkCreated",
+			Status:  metav1.ConditionFalse,
+			Reason:  "SyncError",
+			Message: expectedMessage,
+		}))
+
+		By("verify the first primary UDN remains ready")
+		Eventually(userDefinedNetworkReadyFunc(f.DynamicClient, f.Namespace.Name, firstPrimaryUdnName), 5*time.Second, time.Second).Should(Succeed())
```
Comment (ricky-rav) [inline test/e2e/network_segmentation.go]: we're already checking this on line 1717, so maybe here we can do a `Consistently`?
`Consistently(validateClusterUDNStatusReportsActiveNamespacesFunc(...`

File: test/e2e/network_segmentation.go [hunk 2]
```diff
@@ -1704,6 +1704,46 @@ spec:
...
+
+		By("verify the second primary UDN did not create a NetworkAttachmentDefinition")
+		_, err = nadClient.NetworkAttachmentDefinitions(f.Namespace.Name).Get(context.Background(), secondPrimaryUdnName, metav1.GetOptions{})
```
Comment (ricky-rav) [inline test/e2e/network_segmentation.go]: I would do a `Consistently` check for the lack of NAD

File: test/e2e/network_segmentation.go [hunk 3]
```diff
@@ -2125,6 +2165,64 @@ spec:
...
+			Status:  metav1.ConditionFalse,
+			Reason:  "NetworkAttachmentDefinitionSyncError",
+			Message: expectedMessage,
+		}))
+
+		By("verify the first primary CUDN remains ready")
+		Eventually(clusterUserDefinedNetworkReadyFunc(f.DynamicClient, firstCudnName), 5*time.Second, time.Second).Should(Succeed())
+
+		By("verify the second primary CUDN did not create a NetworkAttachmentDefinition")
+		_, err = nadClient.NetworkAttachmentDefinitions(tenantNamespace).Get(context.Background(), secondCudnName, metav1.GetOptions{})
```
Comment (ricky-rav) [inline test/e2e/network_segmentation.go]: I would do a `Consistently` check for the lack of NAD here too

File: test/e2e/network_segmentation.go [hunk 4]
```diff
@@ -2125,6 +2165,64 @@ spec:
...
+			return normalizeConditions(actualConditions)
+		}, 5*time.Second, 1*time.Second).Should(ContainElement(metav1.Condition{
+			Type:    "NetworkCreated",
+			Status:  metav1.ConditionFalse,
+			Reason:  "NetworkAttachmentDefinitionSyncError",
+			Message: expectedMessage,
+		}))
+
+		By("verify the first primary CUDN remains ready")
+		Eventually(clusterUserDefinedNetworkReadyFunc(f.DynamicClient, firstCudnName), 5*time.Second, time.Second).Should(Succeed())
```
Comment (ricky-rav) [inline test/e2e/network_segmentation.go]: Just like in the UDN test, here I would do a `Consistently` instead of another `Eventually`.

File: test/e2e/network_segmentation.go [hunk 5]
```diff
@@ -2125,6 +2165,64 @@ spec:
...
+
+		By("create first primary CUDN and wait until it is ready")
+		cleanupFirst, err := createManifest(f.Namespace.Name, newPrimaryClusterUDNManifest(cs, firstCudnName, tenantNamespace))
+		DeferCleanup(func() {
+			cleanupFirst()
+			_, err := e2ekubectl.RunKubectl("", "delete", "clusteruserdefinednetwork", firstCudnName, "--wait", fmt.Sprintf("--timeout=%ds", 60))
+			Expect(err).NotTo(HaveOccurred())
+		})
+		Expect(err).NotTo(HaveOccurred())
+		Eventually(clusterUserDefinedNetworkReadyFunc(f.DynamicClient, firstCudnName), 30*time.Second, time.Second).Should(Succeed())
```
Comment (ricky-rav) [inline test/e2e/network_segmentation.go]: Here I think it's more telling if we check for `validateClusterUDNStatusReportsActiveNamespacesFunc`

**Review by ricky-rav — CHANGES_REQUESTED** [review]
Just a few comments to avoid possible flakes :)

File: test/e2e/network_segmentation.go (hunk 1 above)
Comment (jechen0648) [reply test/e2e/network_segmentation.go]: I changed to Consistently, but did not use validateClusterUDNStatusReportsActiveNamespacesFunc(), because I am doing UDN here, at line 2220, I use Consistently(validateClusterUDNStatusReportsActiveNamespacesFunc() for CUDN

Comment (jechen0648) [reply test/e2e/network_segmentation.go]: I wished I thought about keeping the change as a separate commit, I squashed into original first commit, hope it is ok to you :-)

File: test/e2e/network_segmentation.go (hunk 2 above)
Comment (jechen0648) [reply test/e2e/network_segmentation.go]: changed to consistently.   again, I squashed it into first commit before thinking maybe you want to see it as separate commit, sorry about that.

File: test/e2e/network_segmentation.go (hunk 3 above)
Comment (jechen0648) [reply test/e2e/network_segmentation.go]: changed to consistently. again, I squashed it into first commit before thinking maybe you want to see it as separate commit, sorry about that.

File: test/e2e/network_segmentation.go (hunk 4 above)
Comment (jechen0648) [reply test/e2e/network_segmentation.go]: changed to Consistently(validateClusterUDNStatusReportsActiveNamespacesFunc().  again, I squashed it into first commit before thinking maybe you want to see it as separate commit, sorry about that.

File: test/e2e/network_segmentation.go (hunk 5 above)
Comment (jechen0648) [reply test/e2e/network_segmentation.go]: changed to validateClusterUDNStatusReportsActiveNamespacesFunc(), and commited into a separate commit, thanks!

File: test/e2e/network_segmentation.go (hunk 1 above)
Comment (ricky-rav) [reply test/e2e/network_segmentation.go]: yeah yeah, my bad. `validateClusterUDNStatusReportsActiveNamespacesFunc` is clearly only for CUDNs...

Comment (ricky-rav) [reply test/e2e/network_segmentation.go]: No, don't worry about creating extra commits, we usually prefer to have self-contained commits regardless of how many review iterations we do. Fine either way for such small changes. I usually just press on the "compare" button here on github to see the changes the PR author has pushed.

**Review by ricky-rav — APPROVED** [review]
/LGTM

**Review by raphaelvrosa — APPROVED** [review]
